aws_lambda_powertools/metrics/base.py

```diff
@@ -1,77 +1,34 @@
-import datetime
 import functools
 import json
 import logging
-import numbers
 import os
 import warnings
 from collections import defaultdict
 from contextlib import contextmanager
-from enum import Enum
-from typing import Any, Callable, Dict, Generator, List, Optional, Union
+from typing import Any, Callable, Dict, Generator, Optional, Union
 
 from ..shared import constants
 from ..shared.functions import resolve_env_var_choice
-from .exceptions import (
-    MetricResolutionError,
-    MetricUnitError,
-    MetricValueError,
-    SchemaValidationError,
-)
-from .types import MetricNameUnitResolution
+from .exceptions import SchemaValidationError
+from .provider import EMFProvider, MetricsProvider
+from .types import MetricResolution, MetricSummary, MetricUnit
 
 logger = logging.getLogger(__name__)
 
-MAX_METRICS = 100
-MAX_DIMENSIONS = 29
 
 is_cold_start = True
 
 
-class MetricResolution(Enum):
-    Standard = 60
-    High = 1
-
-
-class MetricUnit(Enum):
-    Seconds = "Seconds"
-    Microseconds = "Microseconds"
-    Milliseconds = "Milliseconds"
-    Bytes = "Bytes"
-    Kilobytes = "Kilobytes"
-    Megabytes = "Megabytes"
-    Gigabytes = "Gigabytes"
-    Terabytes = "Terabytes"
-    Bits = "Bits"
-    Kilobits = "Kilobits"
-    Megabits = "Megabits"
-    Gigabits = "Gigabits"
-    Terabits = "Terabits"
-    Percent = "Percent"
-    Count = "Count"
-    BytesPerSecond = "Bytes/Second"
-    KilobytesPerSecond = "Kilobytes/Second"
-    MegabytesPerSecond = "Megabytes/Second"
-    GigabytesPerSecond = "Gigabytes/Second"
-    TerabytesPerSecond = "Terabytes/Second"
-    BitsPerSecond = "Bits/Second"
-    KilobitsPerSecond = "Kilobits/Second"
-    MegabitsPerSecond = "Megabits/Second"
-    GigabitsPerSecond = "Gigabits/Second"
-    TerabitsPerSecond = "Terabits/Second"
-    CountPerSecond = "Count/Second"
-
-
 class MetricManager:
     """Base class for metric functionality (namespace, metric, dimension, serialization)
 
-    MetricManager creates metrics asynchronously thanks to CloudWatch Embedded Metric Format (EMF).
-    CloudWatch EMF can create up to 100 metrics per EMF object
-    and metrics, dimensions, and namespace created via MetricManager
-    will adhere to the schema, will be serialized and validated against EMF Schema.
+    provider refactor: MetricManager implements general Metrics logic, AWS EMF logic are moved to provider.EMFProvider
+
+    MetricManager is now refactored to use provider. Default provider is AWS EMF provider
+    Bring/Implement your own metrics provider to support metrics format other than EMF
```

Comment:
We might need to improve this docstring. 

```suggestion
    MetricManager implements general metrics logic, with providers extending from this class to add specific functionality. Developers can easily implement their own metrics provider by implementing the provider interface. 
    The default provider for the MetricManager is AWS CloudWatch. However, developers can easily bring their own metrics provider to support metrics formats other than EMF.
```

Reply:
Thank you for changing this.